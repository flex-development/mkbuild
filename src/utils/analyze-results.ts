/**
 * @file Utilities - analyzeResults
 * @module mkbuild/utils/analyzeResults
 */

import type { Result } from '#src/interfaces'
import * as color from 'colorette'
import pb from 'pretty-bytes'

/**
 * Returns a pretty printed version of `results`.
 *
 * @param {string} outdir - Output directory name
 * @param {Pick<Result, 'bytes' | 'outfile'>[]} [results=[]] - Build results
 * @param {number} [pad=2] - Number of spaces before new lines
 * @return {string} Pretty printed `results`
 */
const analyzeResults = (
  outdir: string,
  results: Pick<Result, 'bytes' | 'outfile'>[] = [],
  pad: number = 2
): string => {
  /**
   * Total number of bytes in {@link results}.
   *
   * @const {number} bytes
   */
  const bytes: number = results.reduce((acc, result) => acc + result.bytes, 0)

  /**
   * Space before new lines.
   *
   * @const {string} padding
   */
  const padding: string = ' '.repeat(pad)

  return [
    `${padding}${color.bold(outdir)} (total size: ${color.cyan(pb(bytes))})`,
    ...results.map(({ bytes, outfile }) => {
      return color.gray(`${padding}└─ ${outfile} (${pb(bytes)})`)
    })
  ].join('\n')
}

export default analyzeResults
