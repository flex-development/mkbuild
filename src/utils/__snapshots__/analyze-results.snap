// Vitest Snapshot v1

exports[`unit:utils/analyzeResults > should return pretty printed build results 1`] = `
"  [1mdist[22m (total size: [36m22.1 kB[39m)
[90m  â””â”€ dist/index.mjs (30 B)[39m
[90m  â””â”€ dist/index.d.mts (86 B)[39m
[90m  â””â”€ dist/config/constants.mjs (526 B)[39m
[90m  â””â”€ dist/config/constants.d.mts (901 B)[39m
[90m  â””â”€ dist/interfaces/entry.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/entry.d.mts (872 B)[39m
[90m  â””â”€ dist/interfaces/index.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/index.d.mts (342 B)[39m
[90m  â””â”€ dist/interfaces/options.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/options.d.mts (2.29 kB)[39m
[90m  â””â”€ dist/interfaces/result.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/result.d.mts (1.21 kB)[39m
[90m  â””â”€ dist/interfaces/source-file.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/source-file.d.mts (583 B)[39m
[90m  â””â”€ dist/interfaces/statement.mjs (0 B)[39m
[90m  â””â”€ dist/interfaces/statement.d.mts (686 B)[39m
[90m  â””â”€ dist/utils/analyze-results.mjs (575 B)[39m
[90m  â””â”€ dist/utils/analyze-results.d.mts (626 B)[39m
[90m  â””â”€ dist/utils/extract-statements.mjs (1.05 kB)[39m
[90m  â””â”€ dist/utils/extract-statements.d.mts (810 B)[39m
[90m  â””â”€ dist/utils/write.mjs (289 B)[39m
[90m  â””â”€ dist/utils/write.d.mts (619 B)[39m
[90m  â””â”€ dist/plugins/dts/plugin.mjs (2.74 kB)[39m
[90m  â””â”€ dist/plugins/dts/plugin.d.mts (265 B)[39m
[90m  â””â”€ dist/plugins/tsconfig-paths/options.mjs (0 B)[39m
[90m  â””â”€ dist/plugins/tsconfig-paths/options.d.mts (1.04 kB)[39m
[90m  â””â”€ dist/plugins/tsconfig-paths/plugin.mjs (2.56 kB)[39m
[90m  â””â”€ dist/plugins/tsconfig-paths/plugin.d.mts (507 B)[39m
[90m  â””â”€ dist/plugins/fully-specified/options.mjs (0 B)[39m
[90m  â””â”€ dist/plugins/fully-specified/options.d.mts (367 B)[39m
[90m  â””â”€ dist/plugins/fully-specified/plugin.mjs (2.27 kB)[39m
[90m  â””â”€ dist/plugins/fully-specified/plugin.d.mts (878 B)[39m"
`;
